### PR TITLE
include backports_abc

### LINKED ISSUE #44107: salt-ssh 2017.7 doesn't work with Python 3, missing backports_abc
I have a virtualenv Salt setup which I use for salt-ssh, which works perfectly fine with Python 2. However, if I try to run it with Python 3, it fails. I'm running salt-ssh from a Debian 9 machine, the target machine is a Debian 8 machine with Python 3 installed.

```
$ salt-ssh odin test.ping
odin:
    ----------
    retcode:
        0
    stderr:
        Traceback (most recent call last):
          File "/var/tmp/.salt_08831c_salt/py3/tornado/gen.py", line 114, in <module>
            from collections.abc import Generator as GeneratorType  # type: ignore
        ImportError: cannot import name 'Generator'
        
        During handling of the above exception, another exception occurred:
        
        Traceback (most recent call last):
          File "/var/tmp/.salt_08831c_salt/salt-call", line 15, in <module>
            salt_call()
          File "/var/tmp/.salt_08831c_salt/py3/salt/scripts.py", line 386, in salt_call
            import salt.cli.call
          File "/var/tmp/.salt_08831c_salt/py3/salt/cli/call.py", line 9, in <module>
            import salt.cli.caller
          File "/var/tmp/.salt_08831c_salt/py3/salt/cli/caller.py", line 18, in <module>
            import salt.loader
          File "/var/tmp/.salt_08831c_salt/py3/salt/loader.py", line 26, in <module>
            import salt.utils.event
          File "/var/tmp/.salt_08831c_salt/py3/salt/utils/event.py", line 74, in <module>
            import salt.payload
          File "/var/tmp/.salt_08831c_salt/py3/salt/payload.py", line 17, in <module>
            import salt.crypt
          File "/var/tmp/.salt_08831c_salt/py3/salt/crypt.py", line 61, in <module>
            import tornado.gen
          File "/var/tmp/.salt_08831c_salt/py3/tornado/gen.py", line 116, in <module>
            from backports_abc import Generator as GeneratorType  # type: ignore
        ImportError: No module named 'backports_abc'
    stdout:
```

`sudo find /var/tmp/.salt_* -name backports_abc*` on the target system yields no results.

If I manually install backports_abc into the Salt virtualenv, it works again.

```
$ salt-ssh --versions
Salt Version:
           Salt: 2017.7.0-304-gfd2c51b
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.3 (default, Jan 19 2017, 14:11:04)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.1.6
 
System Versions:
           dist: debian 9.1 
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-3-amd64
         system: Linux
        version: debian 9.1
```

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Older version of python need this file to work with tornado when using
salt-ssh.  Since you can run a newer version of python3 that has the file as
collections.abc, it won't be installed, but it still needs to be included in
the thin tarball if the salt-ssh minion is < 3.5

### What issues does this PR fix or reference?
Fixes #44107 

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by terminalmage — APPROVED** [review]

**Comment by rallytime** [comment]
@gtmanfred Lint: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/17949/violations/file/salt/ext/backports_abc.py/

**Comment by gtmanfred** [comment]
Would it be ok if we just disabled pylint fully for this file, because it is a straight copy of the upstream file?

**Comment by thatch45** [comment]
I would say yes, we should disable pylint entirely on the salt/ext directory

**Comment by rallytime** [comment]
Hey @gtmanfred This has a merge conflict now. Pylint is also still unhappy: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/18018/violations/

**Comment by thatch45** [comment]
Since the ext directory was set to be ignored in a separate PR we only need to fix the one parser tests file

File: .pylintrc [hunk 1]
```diff
@@ -11,7 +11,7 @@ profile=no
 
 # Add files or directories to the blacklist. They should be base names, not
 # paths.
-ignore=CVS
+ignore=CVS,salt/ext
```
Comment (rallytime) [inline .pylintrc]: I don't think this change was intended to go here.

File: .testing.pylintrc [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 # Add files or directories to the blacklist. They should be base names, not
 # paths.
-ignore=CVS
+ignore=CVS,ext
```
Comment (rallytime) [inline .testing.pylintrc]: Same comment here as well.

Comment (gtmanfred) [reply .testing.pylintrc]: This one disables running pylint on anything in the ext/ directory as tom mentioned here.

https://github.com/saltstack/salt/pull/45294#issuecomment-356032611

Is there a different way that I could do that?

File: .pylintrc (hunk 1 above)
Comment (gtmanfred) [reply .pylintrc]: yeah, this one can be removed

File: .testing.pylintrc (hunk 2 above)
Comment (rallytime) [reply .testing.pylintrc]: Oh, I am sorry, I missed that comment and got my reviews mixed up. You're right. ++

**Comment by rallytime** [comment]
@gtmanfred There are some ssh tests failing on the Ubuntu runs on this change now:
- integration.ssh.test_state.SSHStateTest.test_state_running
- integration.ssh.test_state.SSHStateTest.test_state_show_top

**Comment by gtmanfred** [comment]
`integration.ssh.test_state.SSHStateTest.test_state_show_top` has gone away and when testing without these patches I still see `integration.ssh.test_state.SSHStateTest.test_state_running` fail

**Comment by gtmanfred** [comment]
looks like test_state_show_top is just going to possibly fail on py3, https://github.com/saltstack/salt-jenkins/issues/686

**Comment by rallytime** [comment]
@gtmanfred Ah, you're right about `test_state_show_top`. But the test failure for `test_state_running` remains: https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/18557/

**Comment by rallytime** [comment]
Ah, I am seeing the`test_state_running` test fail on other PRs so something else is causing that. Sorry for the alarm.

@thatch45 Is this good to go?

**Review by thatch45 — APPROVED** [review]

**Pull request merged**